pkg/cloud/rgraph/exec/executor.go

```diff
@@ -57,6 +59,10 @@ func DryRunOption(dryRun bool) Option {
 	return func(c *ExecutorConfig) { c.DryRun = dryRun }
 }
 
+func TimeoutOption(t time.Duration) Option {
```

Comment:
done

Reply:
In parallel executor yes. I added this in case parallel queue or runAction got stuck somehow.

Reply:
I can add TODO to implement it in serial executor if needed

Reply:
Can this be answered?

Reply:
Please remove this for now until we can get the above question answered.

Reply:
If I did this:

```
ctxWithTimeout := context.WithTimeout(ctx, /*my timeout*/)
ex := NewParallelExecutor()
ex.Run(ctxWithTimeout)
```

Does this do the same time as the `TimeoutOption`? When a parent is canceled -- the children also get canceled as I understand. Then we don't need a special `TimeoutOption`, instead we should document that using `context.WithTimeout` is the way people should be implementing timeouts.

Reply:
TimeoutOption is needed if we wants to wait for all children to finish in case of timeout. There is a function in parallel queue `WaitForOrphans` and it is to ensure that all queue goroutines have returned.  
```
// WaitForOrphans will block until remaining op() goroutines
// finish. Call this if Run() returns an error and you need to know
// that all remaining threads of execution are done.
//
// WaitForOrphans will exit early if ctx is cancelled. This `ctx`
// should be different from the `ctx` given to Run().
```

we cannot handle WaitForOrphans and Run with the same context which is provided by client becasue WaitForOrphans will immediately returns with error. I think that it is a good practice to wait for all go routines to finish or return the error that sth is still running.

Reply:
Ok, let's add WaitForOrphansTimeoutOption as well.

We should also document:

```
// Run executes pending actions in parallel.
//
// ParallelExecutor will stop execution (to the extent possible) if the context
// passed to Run() is cancelled. This will also cancel any waiting for orphan go
// routines that are currently executing.
```

Reply:
done